packages/edit-site/src/components/add-new-template/utils.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { useMemo } from '@wordpress/element';
+
+export const usePostTypes = () => {
+	const postTypes = useSelect(
+		( select ) => select( coreStore ).getPostTypes( { per_page: -1 } ),
+		[]
+	);
+	const excludedPostTypes = [ 'attachment', 'page' ];
+	const filteredPostTypes = postTypes?.filter(
+		( { viewable, slug } ) =>
+			viewable && ! excludedPostTypes.includes( slug )
+	);
+	return filteredPostTypes;
+};
```

Comment:
Based on the diff's comments elsewhere:

> // It's important to use `length` as a dependency because `usePostTypes`
>  // returns a new array every time and will triger a rerender.
>  // We can't avoid that because `post types` endpoint doesn't allow filtering
>  // with `viewable` prop right now.

Why can't `usePostTypes` absorb the issue of reusing object references? A quick sketch:

```suggestion
export const usePostTypes = () => {
	const postTypes = useSelect(
		( select ) => select( coreStore ).getPostTypes( { per_page: -1 } ),
		[]
	);
	return useMemo( () => {
		const excludedPostTypes = [ 'attachment', 'page' ];
		return postTypes?.filter(
			( { viewable, slug } ) =>
				viewable && ! excludedPostTypes.includes( slug )
		);
	}, [ postTypes ] );
};
```

Reply:
Haha, it seems my mind got stuck on this one 😄 . I'll apply this one, thanks!